mcp3008/mcp3008.go

```diff
@@ -0,0 +1,92 @@
+// Package mcp3008 implements a driver for the MCP3008 Analog to Digital Converter.
+//
+// Datasheet: http://ww1.microchip.com/downloads/en/DeviceDoc/21295d.pdf
+//
+package mcp3008 // import "tinygo.org/x/drivers/mcp3008"
+
+import (
+	"errors"
+	"machine"
+)
+
+// Device wraps MCP3008 SPI ADC.
+type Device struct {
+	bus machine.SPI
+	cs  machine.Pin
+	tx  []byte
+	rx  []byte
+	CH0 *ADCPin
+	CH1 *ADCPin
+	CH2 *ADCPin
+	CH3 *ADCPin
+	CH4 *ADCPin
+	CH5 *ADCPin
+	CH6 *ADCPin
+	CH7 *ADCPin
```

Comment:
Why not use an array or slice here?

Also, you could make these non-pointers, which is somewhat lighter on the heap/GC. For example:

```go
type Device struct {
    // ...
    [8]CH ADCPin
}
```

You can always get the pointer with `&device.CH[0]` for example.

Reply:
I was trying to make this act as much as possible like a "regular" `machine.ADC`, hence the specific members. The pointer is so we can in the future use as a param of type `machine.ADConverter`.

Reply:
Got rid of pointers, but left as individual members.